Codec/Archive/Tar/Write.hs

```diff
@@ -59,6 +73,23 @@ putEntry entry = case entryContent entry of
     padding size = LBS.replicate paddingSize 0
       where paddingSize = fromIntegral (negate size `mod` 512)
 
+putEntry' :: GenEntry OsPath TarPath LinkTarget -> IO LBS.ByteString
+putEntry' entry' = do
+  entryContent' <- case entryContent entry' of
+    NormalFile path size -> do
+      content <- defaultRead size path
+      return $ NormalFile content size
```

Comment:
FWIW feel free to use `pure` instead of `return` in all new code, if you like.

Reply:
Do you mean I **SHOULD** use `pure`? I generally use `return` in `do` notation and `pure` otherwise, but I'm not applying that rule consistently for sure.